pandas/core/groupby.py

```diff
@@ -1574,19 +1641,27 @@ def aggregate(self, values, how, axis=0):
         else:
             out_dtype = 'object'
 
-        # will be filled in Cython function
-        result = np.empty(out_shape, dtype=out_dtype)
-        result.fill(np.nan)
-        counts = np.zeros(self.ngroups, dtype=np.int64)
 
-        result = self._aggregate(result, counts, values, agg_func, is_numeric)
+        if kind == 'aggregate':
+            result = np.empty(out_shape, dtype=out_dtype)
```

Comment:
The result is of different shape depending on if it's a transform/aggregation, so it needs to be in the `if`.